File: aiagent7-service/src/main/java/com/shuangqi/aiagent7/utils/rag/FixedWindowTextSplitter.java
package com.shuangqi.aiagent7.utils.rag;

import java.util.ArrayList;
import java.util.List;

public class FixedWindowTextSplitter implements TextSplitter {

    private final int windowSize;

    public FixedWindowTextSplitter(int windowSize) {
        this.windowSize = windowSize;
    }

    @Override
    public List<String> apply(String text) {
        List<String> documents = new ArrayList<>();
        int length = text.length();
        for (int i = 0; i < length; i += windowSize) {
            String chunk = text.substring(i, Math.min(length, i + windowSize));
            documents.add(chunk);
        }
        return documents;
    }


}

File: langhuan-server/src/main/kotlin/com/langhuan/utils/rag/splitter/FixedWindowTextSplitter.kt
package com.langhuan.utils.rag.splitter

class FixedWindowTextSplitter(private val windowSize: Int) : TextSplitter {

    override fun apply(text: String): List<String> {
        val documents = mutableListOf<String>()
        val length = text.length
        for (i in 0 until length step windowSize) {
            val chunk = text.substring(i, minOf(length, i + windowSize))
            documents.add(chunk)
        }
        return documents
    }
}
